Userland/Libraries/LibSoftGPU/SIMD.h

```diff
@@ -103,4 +103,24 @@ ALWAYS_INLINE static Vector2<AK::SIMD::f32x4> ddy(Vector2<AK::SIMD::f32x4> const
     };
 }
 
+// approximation of log2 (https://stackoverflow.com/questions/9411823/fast-log2float-x-implementation-c)
+ALWAYS_INLINE static AK::SIMD::f32x4 log2_approximate(AK::SIMD::f32x4 v)
+{
+    union {
+        AK::SIMD::f32x4 float_val;
+        AK::SIMD::i32x4 int_val;
+    } u { v };
+
+    // extract just the exponent, giving lower integer bound for log2
```

Comment:
Our comment style is to use proper sentences with correct capitalisation and punctuation.

https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#comments